_dasldemos/computational_rank.md

```diff
@@ -0,0 +1,200 @@
+
+There is a graphical user interface to running WDL scripts on your data on `gizmo` called [PROOF](https://proof.fredhutch.org/).
+
+### Containers Option 3
+
+Use Nextflow to run your software. Requires knowledge of Nextflow and Nextflow workflows.
+
+## 3. Best Practices When Using Conda
+
+### Starting Out
+
+1. Conda documentation is [here](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html). If you don’t know where to start, [this page by the Turing Way](https://book.the-turing-way.org/reproducible-research/renv/renv-package) is one of the best resources I can find. 
+2. I recommend installing `miniconda` using the shell script (there are other options to install conda software, including `mamba`). Make sure the `miniconda` directory is added to your `$PATH` variable so you can use the `conda` command. For example, I would alter my `.bashrc` file to contain the line: `export PATH=$PATH:/home/tladera2/miniconda3/bin/` 
+3. Make sure that you’ve updated your conda channels first. To do this, run the following code in your terminal (you only need to do this once, as it creates a configuration file in your home directory).
+
+```bash
+conda config --add channels bioconda
+conda config --add channels conda-forge
+conda config --set channel_priority strict
+```
+
+Note that some channels require a subscription, but `bioconda` and `conda-forge` do not.
+
+### Making a conda environment
+
+1. Use `conda create`  to install your software to the home file system in its own environment. For example: `conda create --name samtools_env samtools=1.19.2-1` will install `samtools` into an environment called `samtools_env`. Try to limit installing more than one utility into each environment to avoid the dependency nightmare.
+
+### In Your Script
+
+1. When using your software, activate your environment using `conda activate` . For example: `conda activate samtools_env` 
+2. Make sure that your software is available using `which` : `which samtools` . This should return a path to where your `samtools` binary is installed.
+3. Do your work in your script with the software.
+4. When you're done with that step, `conda deactivate` the `conda` environment. For example: `conda deactivate samtools_env` 
+
+## 4. Best Practices When Compiling from Source
+
+1. Double check to make sure that there isn’t a container or a precompiled binary available. I’m serious. This is going to take a while, and you want to make sure that you are using your time wisely. 
+2. Ask SciComp if it’s possible to make it into a module (they know a lot!)
+3. Before doing anything: look on Biostars or Stack Overflow to see if there’s compilation advice.
+4. Compile following the instructions. Be prepared to debug a number of steps.
```

Comment:
The compilers on gizmo are very old... in most cases if someone _insists_ on compiling code we suggest loading a core environment module.  There are frequently other dependencies (BLAS is a big one) that are not available in the base OS on the rhino/gizmo nodes.  e.g. `ml spider foss/2023b`

Then they need to be certain to load that module before using the compiled binaries.